src/main/java/com/redhat/jenkins/plugins/ci/CIMessageSubscriberBuilder.java

```diff
@@ -141,27 +121,43 @@ public Integer getTimeout() {
         return timeout;
     }
 
-    @DataBoundSetter
     public void setTimeout(Integer timeout) {
         this.timeout = timeout;
     }
 
+    public ProviderData getProviderData() {
+        return providerData;
+    }
+
+    @DataBoundSetter
+    public void setProviderData(ProviderData providerData) {
+        this.providerData = providerData;
+    }
+
     public JMSMessagingProvider getProvider() {
-        return GlobalCIConfiguration.get().getProvider(providerName);
+        return GlobalCIConfiguration.get().getProvider(providerData.getName());
     }
 
     public String waitforCIMessage(Run<?, ?> build, Launcher launcher, TaskListener listener) {
-        GlobalCIConfiguration config = GlobalCIConfiguration.get();
-        JMSMessagingProvider provider = config.getProvider(providerName);
+        JMSMessagingProvider provider = GlobalCIConfiguration.get().getProvider(providerData.getName());
         if (provider == null) {
             listener.error("Failed to locate JMSMessagingProvider with name "
-                    + providerName + ". You must update the job configuration.");
+                    + providerData.getName() + ". You must update the job configuration.");
             return null;
         }
 
-        JMSMessagingWorker worker =
-                provider.createWorker(overrides, build.getParent().getName());
-        return worker.waitForMessage(build, listener, selector, variable, checks, timeout);
+        if (providerData instanceof ActiveMQSubscriberProviderData) {
```

Comment:
I am wondering here if we can refactor this to reduce the exposure of the concrete classes...

can we not pass the `ProviderData` to the `createWorker()` and let the work do the cast?

This way we reduce the work needed to onboard other providers in the future.